universal-login-react/src/ui/commons/GasPrice.tsx

```diff
@@ -4,28 +4,32 @@ import './../styles/gasPriceDefault.sass';
 import UniversalLoginSDK from '@universal-login/sdk';
 import {utils} from 'ethers';
 import {useAsync} from '../hooks/useAsync';
-import {getGasPriceFor, GasMode} from '@universal-login/commons';
+import {getGasPriceFor, GasMode, GasParameters} from '@universal-login/commons';
 
 interface GasPriceProps {
   sdk: UniversalLoginSDK;
-  setGasTokenAddress: (gasTokenAddress: string) => void;
-  setGasPrice: (gasModeName: utils.BigNumber) => void;
+  onGasParametersChanged: (gasParameters: GasParameters) => void;
 }
 
-export const GasPrice = ({sdk, setGasTokenAddress, setGasPrice}: GasPriceProps) => {
+export const GasPrice = ({sdk, onGasParametersChanged}: GasPriceProps) => {
   const [gasModes] = useAsync(() => sdk.getGasModes(), []);
   const [modeName, setModeName] = useState<string>('');
   const [tokenAddress, setTokenAddress] = useState<string>('');
 
   const onModeChanged = (name: string) => {
     setModeName(name);
-    setGasPrice(getGasPriceFor(gasModes!, name, tokenAddress));
```

Comment:
What if there is no `gasModes`? We should use `ensure(gasModes)`

Reply:
If no gasMode returm Loading

Reply:
My bad